aci/layout.go

```diff
@@ -118,9 +118,16 @@ Tar:
 			}
 			rfsOK = true
 		default:
-			flist = append(flist, name)
+			if _, seen := fseen[name]; seen {
+				return fmt.Errorf("Duplicate file entry in archive: %s", name)
+			}
+			fseen[name] = true
 		}
 	}
+	var flist []string
```

Comment:
Seems ok. You have to pay the cost of decompression as well as reading all of the tar entry headers to know how much you need to allocate. The Golang append complexity is probably well amortized by archive read overhead in a single pass. *waves hands*